cmd/fetchd/cmd/genesis-asi-upgrade.go

```diff
@@ -547,6 +552,84 @@ func replaceAddressInContractStateKey(keyBytes []byte, prefix []byte) string {
 	return hex.EncodeToString(key)
 }
 
+func reconciliationContractStateBalancesRecord(ethAddr string, coins sdk.Coins, networkConfig *NetworkConfig) (*map[string]string, sdk.Int) {
+	amount := coins.AmountOfNoDenomValidation(networkConfig.DenomInfo.OldDenom)
+	if amount.IsZero() {
+		return nil, amount
+	}
+	if amount.IsNegative() {
+		panic(fmt.Errorf("netgative amount value for ethereum '%s' address", ethAddr))
+	}
+
+	var buffer bytes.Buffer
+	writer := bufio.NewWriter(&buffer)
+
+	if _, err := writer.Write(reconciliationBalancesKey); err != nil {
+		panic(err)
+	}
+
+	if ethAddrRaw, err := hex.DecodeString(ethAddr); err != nil {
+		if _, err := writer.Write(ethAddrRaw); err != nil {
+			panic(err)
+		}
+	} else {
+		panic(err)
+	}
```

Comment:
```suggestion
	ethAddrRaw, err := hex.DecodeString(ethAddr)
	if err != nil {
		panic(err)
	}
	if _, err := writer.Write(ethAddrRaw); err != nil {
		panic(err)
	}
```

Reply:
Resoved in the [7a1928a](https://github.com/fetchai/fetchd/pull/365/commits/7a1928ad2ba008ef991066e1099a6645ae1beb9d) commit